### PR TITLE
TST/DEPR: remove .ix from tests\indexing\multiindex\test_loc.py

### PR DESCRIPTION
**Pull request by simonjayhawkins**
follow on from #26438

i leave this as draft until #26438 is merged, to prevent this one being merged first.

not used `iloc` for the expected where dropping levels would be required, used df.values instead.

### CODE REVIEW

**Comment by jreback** [comment]
@simonjayhawkins can you rebase

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 1]
```diff
@@ -84,54 +82,50 @@ def test_loc_getitem_array(self):
...
-                           columns=[[2, 2, 4], [6, 8, 10]],
-                           index=[[4, 4, 8], [8, 10, 12]])
-
-        # the first row
-        rs = mi_labels.loc['i']
-        with catch_warnings(record=True):
-            xp = mi_labels.ix['i']
-        tm.assert_frame_equal(rs, xp)
+        # the first 2 rows
+        expected = DataFrame(
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_loc.py]: use ``.iloc``

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 2]
```diff
@@ -84,54 +82,50 @@ def test_loc_getitem_array(self):
...
+        result = df.loc['i']
+        tm.assert_frame_equal(result, expected)
 
-        # 2nd (last) columns
-        rs = mi_labels.loc[:, 'j']
-        with catch_warnings(record=True):
-            xp = mi_labels.ix[:, 'j']
-        tm.assert_frame_equal(rs, xp)
+        # 2nd (last) column
+        expected = DataFrame(df.values[:, [2]], index=df.index, columns=['B'])
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_loc.py]: same, don't directly use ``.values``

**Review by jreback — CHANGES_REQUESTED** [review]
generally we don't want to use ``DataFrame.values`` instead prefer ``.iloc``

**Comment by simonjayhawkins** [comment]
> generally we don't want to use `DataFrame.values` instead prefer `.iloc`

and then follow with `droplevel`?

**Comment by jreback** [comment]
> and then follow with droplevel?

yes I think the 2nd is more idiomatic

```
In [8]: pd.DataFrame(df.values[[0, 1]], index=['X', 'X'], columns=df.columns)                                                                                                                                                                                                      
Out[8]: 
          i                   j
          A         A         B
X  1.394995 -0.547172  0.790848
X  0.233443 -0.124027  1.298484

In [9]: df.iloc[[0,1]].droplevel(0)                                                                                                                                                                                                                                                
Out[9]: 
          i                   j
          A         A         B
X  1.394995 -0.547172  0.790848
X  0.233443 -0.124027  1.298484

```

**Comment by simonjayhawkins** [comment]
the straight sub fails a couple of tests, i've just commited that to show the failing cases.

next commit using droplevel.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @simonjayhawkins nice clean up!

**Pull request merged**